### ISSUE TITLE
Make walls/texture play nice with ripper94's 3d canvas module

### ISSUE THREAD

**Issue by mcglincy** [mcglincy/dungeondraw-foundry-vtt#22]
https://theripper93.com/

**Comment by brunocalado** [comment]
Hi. I have a simple solution for this.

Add tags to the walls. So, you can read all of them and apply texture and height.

https://github.com/Haxxer/FoundryVTT-Tagger

**Comment by brunocalado** [comment]
I'm using this macro to set all texture and heigth for your module. You could add it for other users.

![image](https://user-images.githubusercontent.com/4999783/145693333-999d9927-acc0-4543-8e24-ba94968b1bed.png)


```js
const texturePath = '3d/textures/wildtextures_grey-tree-bark.jpg';
const wallTopHeight = 10;

let walls = canvas.scene.data.walls.map(w => {
  w = duplicate(w);
  console.log(w)
  
  if (w.flags["levels-3d-preview"]==undefined) {
    w.flags["levels-3d-preview"] = {"wallTexture": ""};
  } 
  if (w.flags["wallHeight"]==undefined) {
    w.flags["wallHeight"] = {"wallHeightTop": wallTopHeight};
  }   
  
  w.flags["levels-3d-preview"].wallTexture = texturePath;
  w.flags.wallHeight.wallHeightTop = wallTopHeight;
  
  return w;
});

canvas.scene.update({walls: walls});
```

**Comment by mcglincy** [comment]
Awesome! Thanks for the code snippet. 

I pinged ripper a couple weeks ago about how to do this and he detailed the flags to add... I just haven't had the cycles to implement it yet. And I need to sign up for his Patreon so I can test it :)

**Comment by brunocalado** [comment]
I understood.

He made a better one.

```js
const texturePath = '3d/textures/wildtextures_grey-tree-bark.jpg';
const doortexturePath = '3d/textures/wildtextures_grey-tree-bark.jpg';
const wallTopHeight = 10;

const walls = canvas.walls.controlled.length ? canvas.walls.controlled : canvas.walls.placeables;

const updates = walls.map(w => {
    const isDoor = w.data.door > 0;
    return {
        _id : w._id,
        flags : {
            "levels-3d-preview" : {
                "wallTexture" : isDoor ? doortexturePath : texturePath
            },
            "wallHeight" : {
                "wallHeightTop" : wallTopHeight
            }
        }
    }
});

canvas.scene.updateEmbeddedDocuments("Wall", updates)
```

**Comment by mcglincy** [comment]
Implemented in upcoming version 0.14.0.
